etc/kayobe/environments/ci-aio/automated-setup.sh

```diff
@@ -2,39 +2,42 @@
 
 set -eux
 
+if [[ ! -f $BASE_PATH/vault-pw ]]; then
+    echo "Vault password file not found at $BASE_PATH/vault-pw"
+    exit 1
+fi
+
 cat << EOF | sudo tee -a /etc/hosts
 10.205.3.187 pulp-server pulp-server.internal.sms-cloud
 EOF
 
-if sudo vgdisplay | grep -q lvm2; then
-   sudo pvresize $(sudo pvs --noheadings | head -n 1 | awk '{print $1}')
-   sudo lvextend -L 4G /dev/rootvg/lv_home -r || true
-   sudo lvextend -L 4G /dev/rootvg/lv_tmp -r || true
-fi
-
 BASE_PATH=~
 KAYOBE_BRANCH=stackhpc/yoga
 KAYOBE_CONFIG_BRANCH=stackhpc/yoga
 
-if [[ ! -f $BASE_PATH/vault-pw ]]; then
-    echo "Vault password file not found at $BASE_PATH/vault-pw"
-    exit 1
+cd $BASE_PATH
+mkdir -p src
+pushd src
+[[ -d kayobe ]] || git clone https://github.com/stackhpc/kayobe.git -b $KAYOBE_BRANCH
+[[ -d kayobe-config ]] || git clone https://github.com/stackhpc/stackhpc-kayobe-config kayobe-config -b $KAYOBE_CONFIG_BRANCH
+popd
+
+if sudo vgdisplay | grep -q lvm2; then
+   sudo pvresize $(sudo pvs --noheadings | head -n 1 | awk '{print $1}')
+   sudo lvextend -L 4G /dev/rootvg/lv_home -r || true
+   sudo lvextend -L 4G /dev/rootvg/lv_tmp -r || true
+else
```

Comment:
I'm not sure exactly at what point it fails, but this was done early to avoid filling the small default LVs.

Reply:
I was hoping for one `if` statement, so wanted this after the kayobe-config clone. I assumed it would be small enough (55MB for kayobe, 26MB for kayobe-config) but it may be too tight. I'll re-arrange.